lib/stages/mergeBeamBuffer.cpp

```diff
@@ -0,0 +1,130 @@
+#include "mergeBeamBuffer.hpp"
+
+#include "BeamMetadata.hpp"   // for BeamMetadata
+#include "StageFactory.hpp"  // for REGISTER_KOTEKAN_STAGE, StageMakerTemplate
+#include "buffer.h"          // for allocate_new_metadata_object, copy_metadata, mark_frame_empty
+#include "chimeMetadata.hpp" // for atomic_add_lost_timesamples, zero_lost_samples
+#include "kotekanLogging.hpp" // for INFO
+
+#include <atomic>      // for atomic_bool
+#include <exception>   // for exception
+#include <functional>  // for _Bind_helper<>::type, bind, function
+#include <regex>       // for match_results<>::_Base_type
+#include <stdexcept>   // for runtime_error
+#include <string>      // for string
+#include <visUtil.hpp> // for frameID, modulo
+#include <iostream>    // for cout
+
+using kotekan::bufferContainer;
+using kotekan::Config;
+using kotekan::Stage;
+using std::string;
+
+REGISTER_KOTEKAN_STAGE(mergeBeamBuffer);
+
+mergeBeamBuffer::mergeBeamBuffer(Config& config_, const std::string& unique_name,
+                               bufferContainer& buffer_container) :
+    Stage(config_, unique_name, buffer_container,
+          std::bind(&mergeBeamBuffer::main_thread, this)) {
+
+    // Apply config
+    _samples_per_data_set = config.get<uint32_t>(unique_name, "samples_per_data_set");
+    _sub_frames_per_merged_frame = config.get<uint32_t>(unique_name, "sub_frames_per_merged_frame"); 
+    in_buf = get_buffer("in_buf");
+    register_consumer(in_buf, unique_name.c_str());
+
+    out_buf = get_buffer("out_buf");
+
+    register_producer(out_buf, unique_name.c_str());
+    
+}
+
+mergeBeamBuffer::~mergeBeamBuffer() {}
+
+
+void mergeBeamBuffer::main_thread() {
+    frameID in_buffer_ID(in_buf);
+    uint8_t* in_frame = nullptr;
+    frameID out_buffer_ID(out_buf);
+    uint8_t* out_frame = nullptr;
+    // Since the input buffer is a ring buffer and the in_buf_ID goes back to zero
+    // when the ID reaches the num_frame, we need something something to keep track
+    // on the incoming frame number
+    uint32_t in_frame_counter = 0;
+    
+    while (!stop_thread) {
+        // Get an input buffer, This call is blocking!
+        in_frame = wait_for_full_frame(in_buf, unique_name.c_str(), in_buffer_ID);
+        if (in_frame == nullptr)
+            break;
+
+	BeamMetadata* in_metadata = (BeamMetadata*)get_metadata(in_buf, in_buffer_ID);
```

Comment:
The indentation in this function makes it a bit hard to read.  You can use `make clang-format` to clean up the indentation if your editor doesn't have a way to do it for you.